### DISCUSSION TITLE
Is there a way to detect prolonged silence?

### DISCUSSION THREAD

**Discussion by hioshih** [llfbandit/record#162] [Q&A] [answered]
Is there a way to detect if the user stopped talking, in order to end the recording without explicitly having him hit a stop button?

Or, can we intercept the recording data in order to implement this somehow?

**Comment by llfbandit** [comment, marked as answer]
There's no such feature. Catching the amplitude is an option (while still inconsistent between platforms) and pause the recording.
But... If paused, there's no more data pushed to this stream.
This could be unlocked though...

In your case, it should be ok-ish...

**Reply by hioshih** [reply]
Thanks, that worked well! Still need to fine tune it a bit, but here is how I tried in the example:

```
int silenceLevel = 0;

  @override
  void initState() {
    _recordSub = _audioRecorder.onStateChanged().listen((recordState) {
      setState(() => _recordState = recordState);
    });

    _amplitudeSub = _audioRecorder
        .onAmplitudeChanged(const Duration(milliseconds: 300))
        .listen((amp) => setState(() {
              _amplitude = amp;
              if ((_amplitude?.current ?? 0) <= -30) {
                silenceLevel++;
                if (silenceLevel > 10) {
                  _stop();
                }
              } else {
                silenceLevel = 0;
              }
              //print('Silence: $silenceLevel Amp: ${_amplitude?.current}');
            }));

    super.initState();
  }
```
